D1673/P1673.md

```diff
@@ -6069,97 +6069,103 @@ public:
     using rank_type = typename extents_type::rank_type;
     using layout_type = layout_blas_packed<Triangle, StorageOrder>;
 
-  private:
-    Extents @_the-extents_@{}; // exposition only
-
-  public:
     constexpr mapping() noexcept = default;
     constexpr mapping(const mapping&) noexcept = default;
-    constexpr mapping(const extents_type& e) noexcept;
+    constexpr mapping(const extents_type&) noexcept;
     template<class OtherExtents>
       constexpr explicit(! is_convertible_v<OtherExtents, extents_type>)
         mapping(const mapping<OtherExtents>& other) noexcept;
 
     constexpr mapping& operator=(const mapping&) noexcept = default;
 
-    constexpr extents_type extents() const noexcept { return @_the-extents_@; }
+    constexpr const extents_type& extents() const noexcept { return @_extents\__@; }
 
-    constexpr size_type required_span_size() const noexcept;
+    constexpr index_type required_span_size() const noexcept;
 
     template<class... Indices>
       constexpr index_type operator() (Indices...) const noexcept;
 
-    static constexpr bool is_always_unique() {
+    static constexpr bool is_always_unique() noexcept {
       return extents_type::static_extent(0) != dynamic_extent &&
         extents_type::static_extent(0) < 2;
     }
-    static constexpr bool is_always_exhaustive() { return true; }
-    static constexpr bool is_always_strided() {
+    static constexpr bool is_always_exhaustive() noexcept { return true; }
+    static constexpr bool is_always_strided() noexcept {
       return extents_type::static_extent(0) != dynamic_extent &&
         extents_type::static_extent(0) < 2;
     }
 
     constexpr bool is_unique() const noexcept {
-      return extent(0) < 2;
+      return @_extents\__@.extent(0) < 2;
     }
     constexpr bool is_exhaustive() const noexcept { return true; }
     constexpr bool is_strided() const noexcept {
-      return extent(0) < 2;
+      return @_extents\__@.extent(0) < 2;
     }
 
     constexpr index_type stride(rank_type) const noexcept;
 
     template<class OtherExtents>
       friend constexpr bool
         operator==(const mapping&, const mapping<OtherExtents>&) noexcept;
+
+  private:
+      extents_type @_extents\__@{}; // exposition only
+  };
 };
 ```
 
-[4]{.pnum} *Constraints:*
+[4]{.pnum} *Mandates:*
 
   * [4.1]{.pnum} `Triangle` is either `upper_triangle_t`
-      or `lower_triangle_t`.
+      or `lower_triangle_t`,
 
   * [4.2]{.pnum} `StorageOrder` is either `column_major_t`
-      or `row_major_t`.
+      or `row_major_t`,
+
+  * [4.3]{.pnum} `Extents` is a specialization of `extents`,
```

Comment:
I like that this is a Mandate.  The wording is not consistent with `layout_left` etc., but I like this wording better anyway (it's a Mandate regardless).

Reply:
Turns out mdspan writes it this way.